### ISSUE TITLE
middlewares support for adapter-node in dev mode.

### ISSUE THREAD

**Issue by Mlocik97** [sveltejs/kit#2325]
### Describe the problem

so there is some basic support for middlewares in adapter-node, but what I saw docs: https://github.com/sveltejs/kit/tree/master/packages/adapter-node#middleware it imports from ./build that exist only in prod mode, but not in dev... 

### Describe the proposed solution

while middlewares should be specific for just adapter-node, wouldn't it be better if that file is exposed from adapter-node import? Like `import { assetsMiddleware, prerenderedMiddleware, kitMiddleware } from '@sveltejs/adapter-node/middlewares';` this would make it work in dev mode and be much better for usage... also docs are little confusing at first look and are "hidden", it would be nice to add it to docs in kit website in adapter section.

### Importance

nice to have

### Additional Information

ofc, we can use vite.server.middlewares but it's duplicating code, and it's also harder to manage, as I can't export middlewares from hooks (as kit doesn't expect it), that I would import in config for vite.

This is little related to: https://github.com/sveltejs/kit/issues/2304

**Comment by benmccann** [comment]
I'm not sure that using stuff from `@sveltejs/adapter-node` in dev mode makes sense, but I agree the APIs are rather different between dev mode and prod mode. I'm hoping to make that better, but have been blocked by getting https://github.com/vitejs/vite/pull/4640 in first. I've pushing to try to get that into tomorrow's release, so fingers crossed

I've been using https://github.com/sveltejs/kit/issues/2304 to track, so will close this as a duplicate of that issue
